### PR TITLE
debug mode, me object and mode ICP

### LINKED ISSUE #5: As a developer I would like to add https support

### LINKED ISSUE #17: As a developer, I would like to register the mode ICP into the backend

### LINKED ISSUE #18: As a developer, I would like to create a me object with the user information and permissions

### LINKED ISSUE #26: As a developer I want to add a debug mode

### LINKED ISSUE #30: As a developer I would like to implement the API end point for the profile page
this will include get and post API end points

### PR DESCRIPTION
**Pull request by ghost**
fixes #26  enable debug mode. to use it you can do npm run debug. This will force the logger to print to the console instead of the log file.
fixes #18 create a me object and have it accessible by the UI to expose some public information about the active session
fixes #17 implement the API to set the mode collab vs class
fixes #30 API end points for fetching and updating profile
fixes #5 enabling https

### CODE REVIEW

File: UI/js/common.js [hunk 1]
```diff
@@ -14,11 +14,33 @@ const errors = Object.freeze({
     2002: 'Invalid username or password',
     2003: 'Invalid username or password',
     2004: 'Invalid username or password',
-    2005: 'Account is not active'
+    2005: 'Account is not active',
+    2006: 'Session timed out',
+    2007: 'failed to update user, missing information'
```
Comment (esteeroramy) [inline UI/js/common.js]: upper case in the beginning of the UI errors please `failed` -> `Failed`

Comment (ahmedqarmout) [reply UI/js/common.js]: nice catch, I will do it for both the common files

File: velocity.js [hunk 2]
```diff
@@ -54,10 +63,10 @@ app.use('/materializecss', express.static(`${__dirname}/node_modules/materialize
 app.use('/animate', express.static(`${__dirname}/node_modules/animate.css/`));
 app.use(
     sassMiddleware({
-        src: `${__dirname}/sass`, 
+        src: `${__dirname}/sass`,
         dest: `${__dirname}/UI/stylesheets`,
-        prefix:  '/stylesheets',
-        debug: true, // TODO: remove before release
+        prefix: '/stylesheets',
+        debug: false, // TODO: remove before release
```
Comment (esteeroramy) [inline velocity.js]: why are we turning this off? lets have it on since we are still developing

Comment (ahmedqarmout) [reply velocity.js]: oooh yeah, sorry forgot to turn it back on. It spams the log files so I turned it off while fixing some issues. But yeah I will turn it back on

**Review by esteeroramy — APPROVED** [review]
minor comments please take a look, nice work

File: Backend/common.js [hunk 3]
```diff
@@ -37,14 +37,23 @@ const errors = Object.freeze({
     1004: 'failed to add user, database issue',
     1005: 'failed to verify password, hashing issue',
     1006: 'failed create user session',
+    1007: 'failed to get settings object, database issue',
+    1008: 'settings object does not exist',
+    1009: 'could not delete the settings object, database issue',
+    1010: 'could not add the settings object, database issue',
+    1011: 'could not update the settings object, database issue',
+    1012: 'could not the selected mode',
```
Comment (SergeyGV) [inline Backend/common.js]: Could not 'what' the selected mode?

Comment (ahmedqarmout) [reply Backend/common.js]: update*, sorry. will fix

**Review by SergeyGV — APPROVED** [review]
Nice work!

**Pull request merged**